pkg/portutil/port_allocate_linux.go

```diff
@@ -0,0 +1,79 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package portutil
+
+import (
+	"fmt"
+
+	"github.com/containerd/nerdctl/pkg/portutil/procnet"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// This port range is compatible with Docker, FYI https://github.com/moby/moby/blob/eb9e42a09ee123af1d95bf7d46dd738258fa2109/libnetwork/portallocator/portallocator_unix.go#L7-L12
+	allocateStart = 49153
+	allocateEnd   = 60999
+)
+
+func filter(ss []procnet.NetworkDetail, filterFunc func(detail procnet.NetworkDetail) bool) (ret []procnet.NetworkDetail) {
+	for _, s := range ss {
+		if filterFunc(s) {
+			ret = append(ret, s)
+		}
+	}
+	return
+}
+
+func portAllocate(protocol string, ip string, count uint64) (uint64, uint64, error) {
+	netprocData, err := procnet.ReadStatsFileData(protocol)
+	logrus.Warn("The file read complete")
```

Comment:
Please remove this when this PR is in mergeable state

Reply:
Marked the PR as a draft

Reply:
I know it, I just test why we always failed in tonistiigi/binfmt now. I will remove it later